test/Elastica/DeprecatedClassBase.php

```diff
@@ -9,15 +9,15 @@
  */
 class DeprecatedClassBase extends Base
 {
-    protected function setUp()
+    protected function set_up()
     {
-        parent::setUp();
+        parent::set_up();
         $this->hideDeprecated();
     }
 
-    protected function tearDown()
```

Comment:
Not sure why we need this change, could you elaborate?
Is it to avoid using the `tearDown(): void`, which needs the `: void` return signature?

Reply:
Yup, exactly. We can't use that on PHP 7.0, as it was added in 7.1.

Dropping 7.0 support would be a breaking change.

Reply:
Dropping php 7.0 would be still a good more IMHO, as it is not supported/maintained from Jan 2019 :)
See: https://www.php.net/supported-versions.php

Reply:
Well, neither is PHP 7.1 or 7.2... ;)

But that's not necessarily my call to make

From https://github.com/ruflin/Elastica/issues/1978#issuecomment-925543910

>I definitively support this. The important part for me is that we don't break existing installations but it seems with this approach this is possible.

So I'm trying to follow that as much as possible.

Now, of course if we change the php version in composer.json, it wouldn't install on the older PHP (and keep using the older version of the 6.x branch), but there's always someone that uses `--ignore-platform-reqs`.

I don't think doing this as a one time change is too much of a maintenance burden.

There is a bigger question of when this branch becomes EOL/also unmaintained, but again, that's not really my call.

Reply:
Oh, and it's not just for 7.0...

As per https://phpunit.de/supported-versions.html PHPUnit 8 supports PHP >= 7.2, meaning that composer will install PHP 8.5 on >= 7.2. And the differene between 6 and 8 brings in those return types.

So yeah, you could make the argument to drop 7.0 and 7.1 based on that (and then remove the old PHPUnit too), but again, breaking changes :).